service/worker/workerdeployment/client.go

```diff
@@ -1206,7 +1208,8 @@ func (d *ClientImpl) updateWithStart(
 
 	policy := backoff.NewExponentialRetryPolicy(100 * time.Millisecond)
 	isRetryable := func(err error) bool {
-		return errors.Is(err, errRetry)
+		// Updates admitted as the workflow is closing may result in a ResourceExhausted error, which is considered retryable.
+		return errors.Is(err, errRetry) || (errors.As(err, new(*serviceerror.ResourceExhausted)) && err.Error() == consts.ErrWorkflowClosing.Error())
```

Comment:
if something like below works it'd be preferable because I don't think we care about the ResourceExhausted part.
```suggestion
		return errors.Is(err, errRetry) || errors.Is(err, consts.ErrWorkflowClosing)
```